pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -169,15 +169,17 @@ func (configgen *ConfigGeneratorImpl) buildClusters(proxy *model.Proxy, req *mod
 	cacheStats := cacheStats{}
 	switch proxy.Type {
 	case model.SidecarProxy:
-		// Setup outbound clusters
-		outboundPatcher := clusterPatcher{efw: envoyFilterPatches, pctx: networking.EnvoyFilter_SIDECAR_OUTBOUND}
-		ob, cs := configgen.buildOutboundClusters(cb, proxy, outboundPatcher, services)
-		cacheStats = cacheStats.merge(cs)
-		resources = append(resources, ob...)
-		// Add a blackhole and passthrough cluster for catching traffic to unresolved routes
-		clusters = outboundPatcher.conditionallyAppend(clusters, nil, cb.buildBlackHoleCluster(), cb.buildDefaultPassthroughCluster())
-		clusters = append(clusters, outboundPatcher.insertedClusters()...)
-
+		// Waypoint proxies do not need outbound clusters
+		if proxy.Metadata.AmbientType != ambient.TypeWaypoint {
```

Comment:
This isn't strictly true. For most cases it is, but if we have a Service or route spanning multiple service accounts, we will need to include outbound cluster in for when we select those.

I am perfectly okay with an MVP not doing this (as long as we filter the endpoints, as well, so we just never send to the other service accounts), but it will be important long term.

Reply:
So we will want, long term, some logic to selectively pick the few that we do need.

Reply:
Actually we might not need these ones, but in `(p *WaypointGenerator) buildClusters(` we will want to remove most of them but add back the few we reference

Reply:
Yup, that's the plan and what I'm working on now: determine how to get what we need and add those specifically.

Reply:
Is it a good idea to have a service spanning multiple accounts?

Reply:
Its not a typical but its a part of Kubernetes so we can't really change it unfortunately

One use case is spanning legacy on prem and K8s, etc

Reply:
Yes, just an example of a Service spanning multiple accounts. Or migrating a service across a namespace. Or to a new service account, etc

Reply:
Isn't there a concept of ownership in k8 Gateway? It seems problematic to have services that are not owned by application developers and can intercept each other.

Reply:
K8s gateway has a strong focus on role boundaries, but doesn't change the semantics of Service that allow a service to select pods belonging to different service accounts.

Reply:
OK, so it implies that waypoints are multi-tenant. But with other things k8s, it's soft and we can ignore noisy neighbor for now.